### ISSUE TITLE
Wrong result for parsing special double number value

### ISSUE THREAD

**Issue by goldmedal** [apache/datafusion-sqlparser-rs#1619]
# Description
I tried to use the nightly commit in the upstream DataFusion (nightly commit) and the tests failed.
- https://github.com/goldmedal/datafusion/pull/2

~~I think it's a bug caused by https://github.com/apache/datafusion-sqlparser-rs/pull/1551~~. Consider the following SQL:
```
select 0. as c1
```
It should be parsed to 
```
ExprWithAlias { 
  expr: Value(Number("0.", false)), 
  alias: Ident { value: "c1", quote_style: None }
}
```
but it seems to be parsed to 
```
CompositeAccess { 
  expr: Value(Number("0", false)), 
  key: Ident { value: "as", quote_style: None, span: Span(Location(0,0)..Location(0,0))
}
```

We should handle this case in `parse_compound_field_access`, too.

**Comment by goldmedal** [comment]
I'll take a look at this issue soon.

**Comment by goldmedal** [comment]
After some tests, I believe this syntax is broken by #1600 and #1598

**Comment by alamb** [comment]
Thanks @goldmedal  and @iffyio 

Do we need to make a patch release for sqlparser 53.0.0? 

By my reading this is an issue that was introduced after that release but before any other release (and thus does not affect the released 53.0.0)

**Comment by goldmedal** [comment]
> Do we need to make a patch release for sqlparser 53.0.0?
>
I think we don't need to do it. 

>By my reading this is an issue that was introduced after that release but before any other release (and thus does not affect the released 53.0.0)
>
Exactly.
